src/Symfony/Component/Notifier/Bridge/Smsbox/SmsboxOptions.php

```diff
@@ -0,0 +1,381 @@
+    {
+        $this->options['dest_iso'] = self::validDestIso($prefixeNumero);
+
+        return $this;
+    }
+
+    public function variable(array $variable)
+    {
+        $this->options['variable'] = $variable;
+
+        return $this;
+    }
+
+    public function coding(string $coding)
+    {
+        $this->options['coding'] = self::validCoding($coding);
+
+        return $this;
+    }
+
+    public function charset(string $charset)
+    {
+        $this->options['charset'] = self::validCharset($charset);
+
+        return $this;
+    }
+
+    public function udh(int $udh)
+    {
+        $this->options['udh'] = self::validUdh($udh);
+
+        return $this;
+    }
+
+    public function callback(bool $callback)
+    {
+        $this->options['callback'] = $callback;
+
+        return $this;
+    }
+
+    public function allowVocal(bool $allowVocal)
+    {
+        $this->options['allow_vocal'] = $allowVocal;
+
+        return $this;
+    }
+
+    public function maxParts(int $maxParts)
+    {
+        $this->options['max_parts'] = self::validMaxParts($maxParts);
+
+        return $this;
+    }
+
+    public function validity(int $validity)
+    {
+        $this->options['validity'] = self::validValidity($validity);
+
+        return $this;
+    }
+
+    public function daysMinMax(int $dayMin, int $dayMax)
+    {
+        $this->options['daysMinMax'] = self::validDays($dayMin, $dayMax);
+
+        return $this;
+    }
+
+    public function hoursMinMax(int $hourMin, int $hourMax)
+    {
+        $this->options['hoursMinMax'] = self::validHours($hourMin, $hourMax);
+
+        return $this;
+    }
+
+    public function sender(string $sender)
+    {
+        $this->options['sender'] = $sender;
+
+        return $this;
+    }
+
+    public function toArray(): array
+    {
+        return $this->options;
+    }
+
+    public static function validMode(string $mode): string
+    {
+        $supportedModes = [
+            self::MESSAGE_MODE_STANDARD,
+            self::MESSAGE_MODE_EXPERT,
+            self::MESSAGE_MODE_RESPONSE,
+        ];
+
+        if (!\in_array($mode, $supportedModes, true)) {
+            throw new InvalidArgumentException(sprintf('The message mode "%s" is not supported; supported message modes are: "%s".', $mode, implode('", "', $supportedModes)));
+        }
+
+        return $mode;
+    }
+
+    public static function validStrategy(int $strategy): int
+    {
+        $supportedStrategies = [
+            self::MESSAGE_STRATEGY_PRIVATE,
+            self::MESSAGE_STRATEGY_NOTIFICATION,
+            self::MESSAGE_STRATEGY_NOT_MARKETING_GROUP,
+            self::MESSAGE_STRATEGY_MARKETING,
+        ];
+        if (!\in_array($strategy, $supportedStrategies, true)) {
+            throw new InvalidArgumentException(sprintf('The message strategy "%s" is not supported; supported strategies types are: "%s".', $strategy, implode('", "', $supportedStrategies)));
+        }
+
+        return $strategy;
```

Comment:
What does the API returns as error if you set `999` for example? Is the error meaningful? If yes, I would avoid client side validation here, because otherwise we would need to update the bridge if a new option is provided by SMSBOX.

Same for the other options

Reply:
Hello, these different options are historical on our API and will not change. We guarantee that there will be no breaking changes. The same applies to our API interface. Thank you

Reply:
I am not talking about changing them, but adding a new one would block the user from using this new option (same for the other options). One would need to make a codechange before on the bridge

Reply:
We won't be adding any new options for the example above, as these are constants based on routing and encoding rules. For example, for SMS routing by operators, which is based on MESSAGE_STRATEGY_*, no other type can exist for carriers.
The same applies to MESSAGE_MODE_*.
And for MESSAGE_CODING_* and MESSAGE_UDH_*, it is based on SMS standards that will never evolve.
 
In addition, the API does not return an error code for the strategy, or the mode provided if it does not exist, the default parameters will be used in this case.

Reply:
Ok then I guess